pkg/client/connection.go

```diff
@@ -25,9 +25,9 @@ package client
 // The connection may consume expensive resources, like TCP connections, or file descriptors, so it
 // is important to reuse it as much as possible, and to close it once it is no longer needed.
 type Connection interface {
-	// Connect creates a ne connection to the messaging broker.
-	Connect() error
-	// Disconnect closes the connection, releasing all the resources that it uses. Once closed the
+	// Open creates a ne connection to the messaging broker.
+	Open() error
```

Comment:
This is OK for now, but I think that eventually the creation/opening of the connection should not be a method in the connection itself, but in some other type (a `ConnectionBuilder`, for example).